src/drt/src/gc/FlexGC_main.cpp

```diff
@@ -1692,6 +1649,181 @@ void FlexGCWorker::Impl::checkMetalShape_minEnclosedArea(gcPin* pin)
   }
 }
 
+void FlexGCWorker::Impl::checkMetalShape_lef58Area(gcPin* pin)
+{
+  if (ignoreMinArea_ || !targetNet_) {
+    return;
+  }
+
+  auto poly = pin->getPolygon();
+  auto layer_idx = poly->getLayerNum();
+  auto layer = getTech()->getLayer(layer_idx);
+
+  if (!layer->hasLef58AreaConstraint()) {
+    return;
+  }
+
+  auto constraints = layer->getLef58AreaConstraints();
+
+  auto sort_cmp =
+      [this](const frLef58AreaConstraint* a, const frLef58AreaConstraint* b) {
+        auto a_rule = a->getODBRule();
+        auto b_rule = b->getODBRule();
+
+        return a_rule->getRectWidth() < b_rule->getRectWidth();
+      };
+
+  // sort constraints to ensure the smallest rect width will have
+  // preference above other rect width statements
+  std::sort(constraints.begin(), constraints.end(), sort_cmp);
+
+  bool check_rect_width = true;
+
+  for (auto con : constraints) {
+    odb::dbTechLayerAreaRule* db_rule = con->getODBRule();
+    auto min_area = db_rule->getArea();
+    auto curr_area = gtl::area(*poly);
+
+    if (curr_area >= min_area) {
+      continue;
+    }
+
+    gtl::rectangle_data<frCoord> bbox;
+    gtl::extents(bbox, *pin->getPolygon());
+    Rect bbox2(gtl::xl(bbox), gtl::yl(bbox), gtl::xh(bbox), gtl::yh(bbox));
+    if (!drWorker_->getDrcBox().contains(bbox2))
+      continue;
+    for (auto& edges : pin->getPolygonEdges()) {
+      for (auto& edge : edges) {
+        if (edge->isFixed())
+          continue;
+      }
+    }
+
+    // add patch only when the poly is not a rect
+    if (checkMetalShape_lef58Area_exceptRectangle(poly, db_rule)) {
+      checkMetalShape_addPatch(pin, min_area);
```

Comment:
checkMetalShape_lef58Area_exceptRectangle returns true if db_rule->isExceptRectangle() is false.  Is it right to patch if you didn't even check?  Should it return false in that case?